nodebuilder/pruner/pruner.go

```diff
@@ -0,0 +1,239 @@
+package pruner
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/ipfs/go-datastore"
+	"github.com/ipfs/go-datastore/namespace"
+	"github.com/ipfs/go-datastore/query"
+	logging "github.com/ipfs/go-log/v2"
+
+	"github.com/celestiaorg/celestia-node/header"
+	"github.com/celestiaorg/celestia-node/share"
+	"github.com/celestiaorg/celestia-node/share/eds"
+)
+
+const (
+	dsPrefix = "/pruner/epoch/"
+)
+
+var log = logging.Logger("pruner")
+
+type StoragePruner struct {
+	cancel context.CancelFunc
+	cfg    Config
+
+	// TODO: Race?
+	oldestEpoch uint64
```

Comment:
i initially thought this was a way we could ship new versions with a known "prune before" timestamp/height, which i realize it isn't, but i wonder if we can/should (ie: like we ship the genesis config, we can set a "pruneBefore" for blocks 30 days before release so new nodes don't have the recency window issue)